index.d.ts

```diff
@@ -164,9 +164,9 @@ export interface Options extends RequestInit {
 	/**
 	Shortcut for sending JSON. Use this instead of the `body` option.
 
-	Accepts a plain object which will be `JSON.stringify()`'d and the correct header will be set for you.
+	Accepts any plain object or value, which will be `JSON.stringify()`'d and sent in the body with the correct header set.
 	*/
-	json?: unknown;
+	json?: object | Array | string | number | boolean | null;
```

Comment:
I am going to undo this change because it is unrelated to supporting non-object JSON values and there is some history behind why the TypeScript type is `unknown`. @sindresorhus made this change and explained it in https://github.com/sindresorhus/ky/pull/133#discussion_r282433699. As far as I know, there still isn't a great solution for that

We can talk about it in the comments here in this PR or you can open a new issue if you want the TypeScript types to be stricter. For now, I think `unknown` is sufficient to support the changes in this PR.

Reply:
I'm :-1: on undoing this. I think Sindre was talking about response, not request. When it comes to parsing response, then yes, you have to cast it on your own because you don't know if it's a string, a number or whatever (one of the types mentioned later in this comment). On the other hand, we don't care about the JSON body as long as it is valid JSON. See, it's valid only when it is `string | number | boolean | null | {[key: string]: ...} | Array<...>`.

@sindresorhus Can you clarify please?

Reply:
Apologies, I linked to the wrong comment. I edited and fixed the link.

Reply:
> Apologies, I linked to the wrong comment. I edited and fixed the link.

No problem.

> And it didn't account for when your have custom types with a toJSON() method.

Well, what about this:

```ts
interface JSONable extends Object {
    toJSON(): string;
}
```

(not sure if it works, not at home atm)

Reply:
`unknown` is the only pragmatic type here. I've tried making it strict before and it just turned out to be an inconvenience to users without much value.